.github/workflows/pr.yml

```diff
@@ -69,7 +69,11 @@ jobs:
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v4
       - name: Build with Gradle
-        run: gradle build --rerun-tasks
+        run: |
+          # fetch submodule tags since actions/checkout@v4 does not
+          git submodule foreach 'git fetch --unshallow || true'
```

Comment:
I haven't tried it but I wonder if the `fetch-tags: true` option for actions/checkout would achieve similar results?